client/src/main/scala/uk/gov/ons/addressIndex/client/AddressIndexClient.scala

```diff
@@ -16,61 +16,72 @@ trait AddressIndexClient {
     * @return the host address of the address index server
     */
   def host: String
-  
-  protected implicit lazy val iClient : WSClient = client
-  protected implicit lazy val iHost : AddressIndexServerHost = AddressIndexServerHost(host)
+
+  protected implicit lazy val iClient: WSClient = client
+  protected implicit lazy val iHost: AddressIndexServerHost = host
 
   /**
     * perform an address search query
     *
-    * @param req the request
+    * @param request the request
     * @return a list of addresses
     */
-  def addressQuery(req : AddressIndexSearchRequest) : Future[_] = {
+  def addressQuery(request: AddressIndexSearchRequest): Future[WSResponse] = {
     AddressQuery
       .toReq
       .withQueryString(
-        Seq(
-          "input" -> req.input,
-          "format" -> req.format.toString
-        )
+        "input" -> request.input,
+        "format" -> request.format.toString
       )
       .get
   }
 
   /**
     * perform a uprn query
     *
-    * @param req the request
+    * @param request the request
     * @return an address
     */
-  def uprnQuery(req : AddressIndexUPRNRequest) : Future[_] = {
-    UprnQuery(req.uprn.toString)
+  def uprnQuery(request: AddressIndexUPRNRequest): Future[WSResponse] = {
+    UprnQuery(request.uprn.toString)
       .toReq
-      .withQueryString(Seq("format" -> req.format.toString))
+      .withQueryString(
+        "format" -> request.format.toString
+      )
       .get
   }
 }
 
-object AddressIndexClient {
+object AddressIndexClientHelper {
 
-  sealed abstract class AddressIndexPath(val path : String, val method : String)
+  implicit def str2host(h: String): AddressIndexServerHost = AddressIndexServerHost(h)
 
-  case class AddressIndexServerHost(value : String)
+  sealed abstract class AddressIndexPath(val path: String, val method: String)
 
-  implicit class AddressIndexPathToWsAugmenter(h : AddressIndexPath)(implicit client : WSClient) {
-    def toReq() : WSRequest = {
+  implicit class AddressIndexPathToWsAugmenter(p: AddressIndexPath)(implicit client: WSClient, host: AddressIndexServerHost) {
+    def toReq(): WSRequest = {
```

Comment:
Ah, fine, just wonder what is the reason
Why "upgrade" class from the outside when you can directly add the method to it?
If you wan't to leave it, leave it